03week/towersOfHanoi.js

```diff
@@ -18,26 +18,58 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
-  // Your code here
-
-}
-
-function isLegal() {
-  // Your code here
-
-}
-
-function checkForWin() {
-  // Your code here
-
+/* Whiteboarding
+ - Create 3 columns(stacks);
+ - Create a variable for 4 disc
+ - Start to movePiece
+- disc1 to stack.b; disc2 to stack.c; disc1 to stack.c; disc 3 to stack.b;
+- disc1 to stack.a; disc2 to stack.b; disc1 to stack.b; disc4 to stack.c;
+- disc1 to stack.c; disc2 to stack.a; disc1 to stack.a; disc3 to stackc;
+- disc1 to stack.b; disc2 to stack.c; disc1 to stack.c for the WIN!.*/
+ */
+
+ const movePiece = (disc,a,b,c) => { //parameter includes 4 disc and 3 columns
+     if (disc > 0) { //if disc are greater than 0(smallest to largest sizes)
+         movePiece(disc - 1,a,c,b); //stack.a will lose the top disc, disc can be moved from all 3 stacks, but the size/value of the disc does not allow larger disc placement over smaller disc.
```

Comment:
your are calling the function inside the function. this would cause an infinite loop.